File: rest-api/src/main/java/com/tecazuay/example/restapi/models/SLA.java
package com.tecazuay.example.restapi.models;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import org.hibernate.annotations.Where;

import com.fasterxml.jackson.annotation.JsonManagedReference;

@Where(clause = "is_deleted = false")
@Entity(name = "sla")
public class SLA extends Globals implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = -9029612685137758093L;
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "sla_id", nullable = false)
	private Long sla_id;

	@Column(name = "tiempo_resolucion", nullable = false, length = 255)
	private String tiempoResolucion;

	@Column(name = "tiempo_respuesta", nullable = false, length = 255)
	private String tiempoRespuesta;

	@Column(name = "reglas_escalada", nullable = false, columnDefinition = "TEXT")
	private String reglasEscalada;

	@JsonManagedReference(value = "rf_sla_criticidad")
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "criticidad_id", nullable = false)
	private Criticidad criticidad;

	@JsonManagedReference(value = "rf_sla_impacto_parametros")
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "impacto_id")
	private Parametros impacto;

	@JsonManagedReference(value = "rf_sla_nivel_prioridad_parametros")
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "nivel_prioridad_id")
	private Parametros nivelPrioridad;

	public Long getSla_id() {
		return sla_id;
	}

	public void setSla_id(Long sla_id) {
		this.sla_id = sla_id;
	}

	public String getTiempoResolucion() {
		return tiempoResolucion;
	}

	public void setTiempoResolucion(String tiempoResolucion) {
		this.tiempoResolucion = tiempoResolucion;
	}

	public String getTiempoRespuesta() {
		return tiempoRespuesta;
	}

	public void setTiempoRespuesta(String tiempoRespuesta) {
		this.tiempoRespuesta = tiempoRespuesta;
	}

	public String getReglasEscalada() {
		return reglasEscalada;
	}

	public void setReglasEscalada(String reglasEscalada) {
		this.reglasEscalada = reglasEscalada;
	}

	public Criticidad getCriticidad() {
		return criticidad;
	}

	public void setCriticidad(Criticidad criticidad) {
		this.criticidad = criticidad;
	}

	public Parametros getImpacto() {
		return impacto;
	}

	public void setImpacto(Parametros impacto) {
		this.impacto = impacto;
	}

	public Parametros getNivelPrioridad() {
		return nivelPrioridad;
	}

	public void setNivelPrioridad(Parametros nivelPrioridad) {
		this.nivelPrioridad = nivelPrioridad;
	}

	public static long getSerialversionuid() {
		return serialVersionUID;
	}
}

File: flutter_app/lib/src/models/sla.dart
import 'package:flutter_app/src/models/criticidad.dart';
import 'package:flutter_app/src/models/parametros.dart';

class Sla {
  late String createdAt;
  late String updatedAt;
  late int slaId;
  late String tiempoResolucion;
  late String tiempoRespuesta;
  late String reglasEscalada;
  late Criticidad criticidad;
  late ParametroModel impacto;
  late ParametroModel nivelPrioridad;

  Sla({
    required this.createdAt,
    required this.updatedAt,
    required this.slaId,
    required this.tiempoResolucion,
    required this.tiempoRespuesta,
    required this.reglasEscalada,
    required this.criticidad,
    required this.impacto,
    required this.nivelPrioridad,
  });

  Sla.fromJson(Map<String, dynamic> json) {
    createdAt = json['createdAt'];
    updatedAt = json['updatedAt'];
    slaId = json['sla_id'];
    tiempoResolucion = json['tiempoResolucion'];
    tiempoRespuesta = json['tiempoRespuesta'];
    reglasEscalada = json['reglasEscalada'];
    criticidad = new Criticidad.fromJson(json['criticidad']);
    impacto = new ParametroModel.fromJson(json['impacto']);
    nivelPrioridad = new ParametroModel.fromJson(json['nivelPrioridad']);
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['createdAt'] = this.createdAt;
    data['updatedAt'] = this.updatedAt;
    data['sla_id'] = this.slaId;
    data['tiempoResolucion'] = this.tiempoResolucion;
    data['tiempoRespuesta'] = this.tiempoRespuesta;
    data['reglasEscalada'] = this.reglasEscalada;
    data['criticidad'] = this.criticidad.toJson();
    data['impacto'] = this.impacto.toJson();
    data['nivelPrioridad'] = this.nivelPrioridad.toJson();

    return data;
  }
}
